src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -0,0 +1,135 @@
+                {
+                    Console.WriteLine($"No channels found with name containing '{_options.Channel}'");
+                    Console.WriteLine("Available channels:");
+                    foreach (Channel channel in channels)
+                    {
+                        Console.WriteLine($"  {channel.Name}");
+                    }
+                    return Constants.ErrorCode;
+                }
+                else if (matchingChannels.Count != 1)
+                {
+                    Console.WriteLine($"Multiple channels found with name containing '{_options.Channel}', please select one");
+                    foreach (Channel channel in matchingChannels)
+                    {
+                        Console.WriteLine($"  {channel.Name}");
+                    }
+                    return Constants.ErrorCode;
+                }
+
+                // Find the build to give someone info
+                Build build = await remote.GetBuildAsync(_options.Id);
+                if (build == null)
+                {
+                    Console.WriteLine($"Could not find a build with id '{_options.Id}'");
+                    return Constants.ErrorCode;
+                }
+
+                Channel targetChannel = matchingChannels.Single();
+
+                if (build.Channels.Any(c => c.Id == targetChannel.Id))
+                {
+                    Console.WriteLine($"Build '{build.Id}' has already been assigned to '{targetChannel.Name}'");
+                    return Constants.SuccessCode;
+                }
+
+                Console.WriteLine($"Assigning the following build to channel '{targetChannel.Name}':");
+                Console.WriteLine();
+                OutputFormattingHelpers.PrintBuild(build);
+
+                await remote.AssignBuildToChannel(_options.Id, targetChannel.Id);
+
+                // Be helpful. Let the user know what will happen.
+                string buildRepo = build.GitHubRepository ?? build.AzureDevOpsRepository;
+                List<Subscription> applicableSubscriptions = (await remote.GetSubscriptionsAsync(
+                    sourceRepo: buildRepo, channelId: targetChannel.Id)).ToList();
+                IEnumerable<Subscription> subscriptionsThatWillFlowImmediately = applicableSubscriptions.Where(s => s.Enabled &&
+                    s.Policy.UpdateFrequency == SubscriptionPolicyUpdateFrequency.EveryBuild);
+                IEnumerable<Subscription> subscriptionsThatWillFlowTomorrowOrNotAtAll = applicableSubscriptions.Where(s => s.Enabled &&
+                    (s.Policy.UpdateFrequency == SubscriptionPolicyUpdateFrequency.EveryDay ||
+                    s.Policy.UpdateFrequency == SubscriptionPolicyUpdateFrequency.None));
+                IEnumerable<Subscription> disabledSubscriptions = applicableSubscriptions.Where(s => !s.Enabled);
+
+                // Print out info
+                if (subscriptionsThatWillFlowImmediately.Any())
```

Comment:
Maybe interesting to factor out the body of the next 3 if blocks to a method. With some changes to the labels.